clusterMembership/clusterMembership.go

```diff
@@ -0,0 +1,21 @@
+package clusterMembership
+
+
+type Node interface {
+	GetId() string
+	SendMessage(msg string) error
+}
+
+type Cluster interface {
+	/*
+	 * Returns a Slice of Node Ids in the Cluster
+	 */
+	GetMembers() []string
+
+	/*
+	 * Sends a Message to the specified Node in the Cluster,
+	 * Returns an Error if the Node is not in the cluster or sending
+	 * the message fails
+	 */
+	SendMessage(msg string, nodeId string) (error)
```

Comment:
Yes different Implementations of the Node interface are how we define a worker that exposes a thrift api or http endpoint for example if its on a different machine.

We can also have different implementations of dynamic cluster membership, one that relies on ZK for update or dynamic config files, or even a gossip protocol for example